### PR TITLE
windowsdesktop-runtime: Add version 3.1.5

### PR DESCRIPTION
**Pull request by x-magic**
Added .NET Core Windows Desktop Runtime version 3.1.5

Runtime installer just like other [vcredists](https://github.com/lukesampson/scoop-extras/blob/master/bucket/vcredist.json), but used by some .net core apps (like [PowerToys](https://github.com/microsoft/PowerToys))

### CODE REVIEW

File: bucket/dotnet-core-desktop-runtime.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "homepage": "https://dotnet.microsoft.com/download/dotnet-core",
+    "description": "Microsoft .NET Core Desktop Runtime",
+    "version": "3.1.5",
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:4]:
```suggestion
    "version": "3.1.5",
    "description": "Microsoft .NET Core Desktop Runtime",
    "homepage": "https://dotnet.microsoft.com/download/dotnet-core",
```

File: bucket/dotnet-core-desktop-runtime.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x64.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
+        },
+        "32bit": {
+            "url": "https://download.visualstudio.microsoft.com/download/pr/df7b90d9-b93e-4974-85ef-c1de418bc186/e380e58bbd8505ebaee6c3abb23baade/windowsdesktop-runtime-3.1.5-win-x86.exe",
+            "hash": "sha512:b5e7e49d5b6f4f83a665d1f4c7d83700540f9c888c3c36ccd44f5395138dc1e258a4fcb260845d8ac7e3e5a1513be1666521edeaf6dfd67669a04a1290cda7e1",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x86.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
+        }
+    },
+    "notes": "You can now remove this installer with 'scoop uninstall dotnet-core-desktop-runtime'"
+}
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:22]:
```suggestion
}

```

File: bucket/dotnet-core-desktop-runtime.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            "hash": "sha512:5df17bd9fed94727ec5b151e1684bf9cdc6bfd3075f615ab546759ffca0679d23a35fcf7a8961ac014dd5a4ff0d22ef5f7434a072e23122d5c0415fcd4198831",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x64.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
+        },
+        "32bit": {
+            "url": "https://download.visualstudio.microsoft.com/download/pr/df7b90d9-b93e-4974-85ef-c1de418bc186/e380e58bbd8505ebaee6c3abb23baade/windowsdesktop-runtime-3.1.5-win-x86.exe",
+            "hash": "sha512:b5e7e49d5b6f4f83a665d1f4c7d83700540f9c888c3c36ccd44f5395138dc1e258a4fcb260845d8ac7e3e5a1513be1666521edeaf6dfd67669a04a1290cda7e1",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x86.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
+        }
+    },
+    "notes": "You can now remove this installer with 'scoop uninstall dotnet-core-desktop-runtime'"
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:21]:
```suggestion
    },
    "pre_install": "if (!(is_admin)) { error 'Admin privileges are required.'; break }",
    "installer": {
        "script": "Invoke-ExternalCommand \"$dir\\$fname\" -ArgumentList '/install', '/quiet', '/norestart' -RunAs | Out-Null"
    }
````

File: bucket/dotnet-core-desktop-runtime.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+    "homepage": "https://dotnet.microsoft.com/download/dotnet-core",
+    "description": "Microsoft .NET Core Desktop Runtime",
+    "version": "3.1.5",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://www.microsoft.com/en-us/legal/intellectualproperty/copyright/default.aspx"
+    },
+    "architecture": {
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:9]:
```suggestion
    "notes": "Installed files could be removed with 'scoop uninstall dotnet-core-desktop-runtime'",
    "architecture": {
```

File: bucket/dotnet-core-desktop-runtime.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        "64bit": {
+            "url": "https://download.visualstudio.microsoft.com/download/pr/86835fe4-93b5-4f4e-a7ad-c0b0532e407b/f4f2b1239f1203a05b9952028d54fc13/windowsdesktop-runtime-3.1.5-win-x64.exe",
+            "hash": "sha512:5df17bd9fed94727ec5b151e1684bf9cdc6bfd3075f615ab546759ffca0679d23a35fcf7a8961ac014dd5a4ff0d22ef5f7434a072e23122d5c0415fcd4198831",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x64.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
+        },
+        "32bit": {
+            "url": "https://download.visualstudio.microsoft.com/download/pr/df7b90d9-b93e-4974-85ef-c1de418bc186/e380e58bbd8505ebaee6c3abb23baade/windowsdesktop-runtime-3.1.5-win-x86.exe",
+            "hash": "sha512:b5e7e49d5b6f4f83a665d1f4c7d83700540f9c888c3c36ccd44f5395138dc1e258a4fcb260845d8ac7e3e5a1513be1666521edeaf6dfd67669a04a1290cda7e1",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x86.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:18]: Do not use sha512 when you do not have hash extraction in place

```suggestion
            "hash": ""
```

File: bucket/dotnet-core-desktop-runtime.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "3.1.5",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://www.microsoft.com/en-us/legal/intellectualproperty/copyright/default.aspx"
+    },
+    "architecture": {
+        "64bit": {
+            "url": "https://download.visualstudio.microsoft.com/download/pr/86835fe4-93b5-4f4e-a7ad-c0b0532e407b/f4f2b1239f1203a05b9952028d54fc13/windowsdesktop-runtime-3.1.5-win-x64.exe",
+            "hash": "sha512:5df17bd9fed94727ec5b151e1684bf9cdc6bfd3075f615ab546759ffca0679d23a35fcf7a8961ac014dd5a4ff0d22ef5f7434a072e23122d5c0415fcd4198831",
+            "post_install": "Invoke-ExternalCommand -FilePath \"$dir\\windowsdesktop-runtime-3.1.5-win-x64.exe\" -ArgumentList \"/install /quiet /norestart\" -RunAs | Out-Null"
```
Comment (Ash258) [inline bucket/dotnet-core-desktop-runtime.json:13]:
```suggestion
            "hash": "",
```

File: bucket/dotnet-core-desktop-runtime.json (hunk 5 above)
Comment (x-magic) [reply bucket/dotnet-core-desktop-runtime.json:18]: I can only find references to hash extraction in [this Wiki article](https://github.com/lukesampson/scoop/wiki/App-Manifest-Autoupdate). Is this applicable to an app json manifest as well? 

MS website provides only sha512 hash on the [download page](https://dotnet.microsoft.com/download/dotnet-core/thank-you/runtime-desktop-3.1.5-windows-x64-installer), and I haven't managed to find a json descriptor that contains everything. If that's the case, do you prefer to have no hash at all?

File: bucket/dotnet-core-desktop-runtime.json (hunk 3 above)
Comment (x-magic) [reply bucket/dotnet-core-desktop-runtime.json:21]: Usually speaking a UAC window will pop up and ask for admin rights anyway. I assume if pre_install condition is added, and user didn't execute the install command with sudo or from escalated shell, they will need to do "scoop uninstall" then "scoop install"?

Comment (Ash258) [reply bucket/dotnet-core-desktop-runtime.json:21]: ...

https://scoop.sh/

> Say goodbye to permission pop-ups
> Scoop installs programs to your home directory by default. So you don’t need admin permissions to install programs, and you won’t see UAC popups every time you need to add or remove a program.

**Comment by x-magic** [comment]
> I would keep it consistent with MS `windesktop-runtime.json`
> 
> Fixed manifest

Haha I was thinking of naming it windowsdesktop-runtime in the beginning but was worried about ambiguity. Yeah I agree to name it after MS. I'll update this PR now.

**Comment by Ash258** [comment]
Do user need both x86 and x64 to be installed?

**Comment by x-magic** [comment]
> Do user need both x86 and x64 to be installed?

No, only platform-specific one need to be installed.

**Comment by x-magic** [comment]
Hi, is there any other feedback pending changes? Please let me know.

**Pull request merged**